src/main/java/com/mongodb/kafka/connect/util/ConfigHelper.java

```diff
@@ -191,18 +192,17 @@ public static Optional<ConfigValue> getConfigByNameWithoutErrors(
     return Optional.empty();
   }
 
-  public static Config evaluateConfigValues(
-      final Config rawConfig, final AbstractConfig resolvedConfig) {
-    final Map<String, ?> evalValues = resolvedConfig.values();
-    rawConfig
-        .configValues()
-        .forEach(
-            configValue -> {
-              Object ev = evalValues.get(configValue.name());
-              if (ev != null) {
-                configValue.value(ev);
-              }
-            });
-    return rawConfig;
+  public static Map<String, String> evaluateConfigValues(
+      final Map<String, String> rawConfigs, final AbstractConfig resolvedConfig) {
+    Map<String, String> resolvedRawConfigs = new HashMap<>(rawConfigs);
+    final Map<String, Object> originals = resolvedConfig.originals();
```

Comment:
@jagadishmdb 
The other solution I can think of is change this line #198 to:
`final Map<String, ?> originals = resolvedConfig.values();`
I can do some testing with this tomorrow. I think this is going to be better than getting the originals.

Reply:
I did some analysis with this approach but I think the previous solution was better. values property are typecasted based on ConfigDef. So there is a chance something may be ignored due to type issue in future.